core/dbt/adapters/sql/connections.py

```diff
@@ -128,6 +132,14 @@ def get_result_from_cursor(cls, cursor: Any) -> agate.Table:
 
         return dbt.clients.agate_helper.table_from_data_flat(data, column_names)
 
+    @classmethod
+    def data_type_code_to_name(cls, type_code: Union[int, str]) -> str:
+        """Get the string representation of the data type from the type_code."""
+        # https://peps.python.org/pep-0249/#type-objects
+        raise dbt.exceptions.NotImplementedError(
+            "`data_type_code_to_name` is not implemented for this adapter!"
+        )
+
```

Comment:
I was interested in seeing a tangible example of how data type codes map to a string representation for a database connector.

This table from the Snowflake docs was useful to me:
https://docs.snowflake.com/en/user-guide/python-connector-api#label-python-connector-type-codes

| type_code | String Representation | Data Type      |
|-----------|-----------------------|----------------|
| 0         | FIXED                 | NUMBER/INT     |
| 1         | REAL                  | REAL           |
| 2         | TEXT                  | VARCHAR/STRING |
| 3         | DATE                  | DATE           |
| 4         | TIMESTAMP             | TIMESTAMP      |
| 5         | VARIANT               | VARIANT        |
| 6         | TIMESTAMP_LTZ         | TIMESTAMP_LTZ  |
| 7         | TIMESTAMP_TZ          | TIMESTAMP_TZ   |
| 8         | TIMESTAMP_NTZ         | TIMESTAMP_TZ   |
| 9         | OBJECT                | OBJECT         |
| 10        | ARRAY                 | ARRAY          |
| 11        | BINARY                | BINARY         |
| 12        | TIME                  | TIME           |
| 13        | BOOLEAN               | BOOLEAN        |

(Side note: I _suspect_ there is a typo for code 8 and `TIMESTAMP_TZ` there should be `TIMESTAMP_NTZ` instead.)